### PR TITLE
SALTO-4266: delete empty accountIds in scriptRunner

### PR DESCRIPTION
**Pull request by IdoZakk**
_Removed empty accountIds to avoid errors during deployment_

---

_Not adding noise reduction as we do not have customers with enabled scriptrunner, except maybe Convex_

---
_Release Notes_: 
Jira Adapter:
* Fixed a bug in scriptRunner that prevented deployment of notification post functions without users

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/jira-adapter/src/filters/script_runner/workflow/empty_account_ids.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import { isInstanceElement, Element, isAdditionOrModificationChange, getChangeData, isInstanceChange, Value } from '@salto-io/adapter-api'
+import { collections } from '@salto-io/lowerdash'
+import { FilterCreator } from '../../../filter'
+import { WorkflowInstance, isWorkflowInstance } from '../../workflow/types'
+import { SCRIPT_RUNNER_POST_FUNCTION_TYPE, SCRIPT_RUNNER_SEND_NOTIFICATIONS } from './workflow_cloud'
+
+const { makeArray } = collections.array
+
+const changeAccountIds = (func: (scriptRunner: Value) => Value):
+((workflowInstance: WorkflowInstance) => void) => (workflowInstance: WorkflowInstance): void => {
```
Comment (alonstern) [inline packages/jira-adapter/src/filters/script_runner/workflow/empty_account_ids.ts:26]: These are pretty hard to follow, can it not be a function that returns a function and call it like `changeAccountIds(deleteEmptyAccountsId, workflowInstance)`?

File: packages/jira-adapter/src/filters/script_runner/workflow/empty_account_ids.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  workflowInstance.value.transitions.forEach(transition => {
+    makeArray(transition.rules?.postFunctions).forEach(postFunction => {
+      if (postFunction.type === SCRIPT_RUNNER_POST_FUNCTION_TYPE
+          && postFunction.configuration?.scriptRunner !== undefined
+          && postFunction.configuration.scriptRunner.className === SCRIPT_RUNNER_SEND_NOTIFICATIONS) {
```
Comment (alonstern) [inline packages/jira-adapter/src/filters/script_runner/workflow/empty_account_ids.ts:31]: nit:
```suggestion
          && postFunction.configuration.scriptRunner?.className === SCRIPT_RUNNER_SEND_NOTIFICATIONS) {
```

Comment (alonstern) [inline packages/jira-adapter/src/filters/script_runner/workflow/empty_account_ids.ts:31]: Are we sure `SCRIPT_RUNNER_SEND_NOTIFICATIONS` is the only case of this? I think in that sense ignoring `""` in the change validator would be safer, but what ever you think

Comment (IdoZakk) [reply packages/jira-adapter/src/filters/script_runner/workflow/empty_account_ids.ts:31]: The change validator is based on the general account IDs function `accountIdsScenarios`
I did not want to add a specific case just for that. And always ignoring "" is not valid as in most cases you must specify the user
